src/ert/analysis/_es_update.py

```diff
@@ -175,66 +178,49 @@ def _param_ensemble_for_projection(
             (
-                temp_storage[parameter.name][parameter.index_list, :].astype(np.double),
+                da.values.T[parameter.index_list, :].astype(np.double),
                 parameter.row_scaling,
             ),
         )
 
     return matrices
 
 
-def _save_to_temp_storage(
-    temp_storage: TempStorage,
-    parameter: RowScalingParameter,
-    A: Optional[npt.NDArray[np.double]],
-) -> None:
-    if A is None:
-        return
-    active_indices = parameter.index_list
-    if active_indices is None:
-        temp_storage[parameter.name] = A
-    else:
-        temp_storage[parameter.name][active_indices, :] = A[active_indices, :]
-
-
 def _save_temp_storage_to_disk(
     target_fs: EnsembleAccessor,
-    temp_storage: TempStorage,
+    da: xr.DataArray,
     iens_active_index: npt.NDArray[np.int_],
+    param_group: str,
 ) -> None:
-    for key, matrix in temp_storage.items():
-        config_node = target_fs.experiment.parameter_configuration[key]
-        for i, realization in enumerate(iens_active_index):
-            if isinstance(config_node, GenKwConfig):
-                assert isinstance(matrix, np.ndarray)
-                dataset = xr.Dataset(
-                    {
-                        "values": ("names", matrix[:, i]),
-                        "transformed_values": (
-                            "names",
-                            config_node.transform(matrix[:, i]),
-                        ),
-                        "names": [e.name for e in config_node.transfer_functions],
-                    }
-                )
-                target_fs.save_parameters(key, realization, dataset)
-            elif isinstance(config_node, (Field, SurfaceConfig)):
-                _matrix = temp_storage.get_xr_array(key, i)
-                assert isinstance(_matrix, xr.DataArray)
-                target_fs.save_parameters(key, realization, _matrix.to_dataset())
-            else:
-                raise NotImplementedError(f"{type(config_node)} is not supported")
+    config_node = target_fs.experiment.parameter_configuration[param_group]
+    for i, realization in enumerate(iens_active_index):
+        if isinstance(config_node, GenKwConfig):
+            dataset = xr.Dataset(
+                {
+                    "values": ("names", da.values.T[:, i]),
+                    "transformed_values": (
+                        "names",
+                        config_node.transform(da.values.T[:, i]),
+                    ),
+                    "names": [e.name for e in config_node.transfer_functions],
+                }
+            )
+            target_fs.save_parameters(param_group, realization, dataset)
+        elif isinstance(config_node, (Field, SurfaceConfig)):
+            target_fs.save_parameters(
+                param_group, realization, da.sel(realizations=realization).to_dataset()
+            )
+        else:
+            raise NotImplementedError(f"{type(config_node)} is not supported")
```

Comment:
Think we can safely get rid of this, as it is quite unlikely that another node will get in here 😅  Probably a remnant from when all nodes were in the same list, responses and parameters.